File: cs/Binder.js
function Binder(editor) {
	var _ = this._ = editor;

	var init = false;
	this.initAfterLoad = function() {
		if (this.init) return;
		this.init = true;
		_.initAfterLoad();
	}
	this.setMenus = function(menus) {
		_.setMenus(menus);
	}
	
	this.moveWindow = function(name, x, y, w, h, resize) {
		_.moveWindow(name, x, y, w, h, resize);
	}
	this.focus = function(target) {
		_.focusWindow(target);
	}
	this.setFollowWindow = function(follow) {
		_.setFollowWindow(follow);
	}

	this.showDragging = function(id) {
		_.showDragging(id);
	}
	this.hideDragging = function() {
		_.hideDragging();
	}
	
	this.focusToMenu = function(keyCode) {
		//_.focusToMenu(keyCode);
	}

	this.saveSetting = function(setting) {
		_.saveSetting(setting);
	}
	this.setVideoExts = function(exts) {
		_.setVideoExts(exts);
	}
	this.setPlayer = function(dll, exe, withRun) {
		_.setPlayer(dll, exe, withRun);
	}
	this.runPlayer = function(path) {
		_.runPlayer(path);
	}

	this.save = function(text, path) {
		_.save(text, path);
	}
	this.saveTemp = function(text, path) {
		_.saveTemp(text, path);
	}
	this.openFile = function() {
		_.openFile();
	}
	this.openFileForVideo = function() {
		_.openFileForVideo();
	}
	this.checkLoadVideoFile = function(smiPath) {
		_.checkLoadVideoFile(smiPath);
	}
	this.loadVideoFile = function(path) {
		_.loadVideoFile(path);
	}
	this.openTempDir = function() {
		_.openTempDir();
	}
	this.doExit = function(resetPlayer, exitPlayer) {
		_.doExit(resetPlayer, exitPlayer);
	}
	
	// 팝업 통신
	this.sendMsg = function(target, msg) {
		_.sendMsg(target, msg);
	}
	
	// setting.html
	this.getWindows = function(targets) { _.getWindows(targets); }
	this.selectPlayerPath = function() { alert("C#에서 동작합니다."); }
	
	// addon 설정 용
	this.loadAddonSetting = function(path) { _.loadAddonSetting(path); }
	this.saveAddonSetting = function(path, text) { _.saveAddonSetting(path, text); }
	
	// viewer/finder opener 못 쓰게 될 경우 고려
	this.updateViewerSetting  = function() { _.updateViewerSetting(); };
	this.updateViewerLines = function(lines) { _.updateViewerLines(lines); };
	
	this.onloadFinder  = function(last  ) { _.onloadFinder (last  ); };
	this.runFind	   = function(params) { _.runFind      (params); };
	this.runReplace    = function(params) { _.runReplace   (params); };
	this.runReplaceAll = function(params) { _.runReplaceAll(params); };

	this.alert   = function(target, msg) { _.alert  (target, msg); }
	this.confirm = function(target, msg) { _.confirm(target, msg); }
	this.prompt  = function(target, msg) { _.prompt (target, msg); }

	// 플레이어
	this.playOrPause = function() { _.player.playOrPause(); }
	this.play        = function() { _.player.play(); }
	this.stop        = function() { _.player.stop(); }
	this.moveTo  = function(time) { _.player.moveTo(time); }

	// 부가기능
	this.runColorPicker = function() { alert("C#에서 동작합니다."); }
	this.normalize = function(text) {
		var smi = new Subtitle.SmiFile();
		var input = smi.fromTxt(text).body;
		Subtitle.Smi.normalize(input);
		smi.body = input;
		output = smi.toTxt().trim();
		_.afterTransform(output);
	}
	this.fillSync = function(text) {
		var smi = new Subtitle.SmiFile();
		var input = smi.fromTxt(text).body;
		Subtitle.Smi.fillEmptySync(input);
		smi.body = input;
		output = smi.toTxt().trim();
		_.afterTransform(output);
	}
	this.toAss = function(text, afterFunc) {
		_.toAss(text, afterFunc);
	}
};

File: Jamaker/Binder.cs
using Newtonsoft.Json;
using System;

namespace Jamaker
{
    class BaseBinder
    {
        private readonly MainForm _;

        public static T ParseJson<T>(string input)
        {
            return JsonConvert.DeserializeObject<T>(input);
        }

        public BaseBinder(MainForm webForm)
        {
            _ = webForm;
        }

        public void Focus(string target)
        {
            _.FocusWindow(target);
        }

        public void InitAfterLoad(string title)
        {
            _.InitAfterLoad(title);
        }

        public void ShowDragging()
        {
            _.ShowDragging();
        }
        public void HideDragging()
        {
            _.HideDragging();
        }

        public void Alert(string target, string msg) { _.Alert(target, msg); }
        public void Confirm(string target, string msg) { _.Confirm(target, msg); }
        public void Prompt(string target, string msg, string def) { _.Prompt(target, msg, def); }
    }

    class Binder : BaseBinder
    {
        private readonly MainForm _;

        public Binder(MainForm mainForm) : base(mainForm)
        {
            _ = mainForm;
        }

        public void MoveWindow(string target, int x, int y, int width, int height, bool resizable)
        {
            _.MoveWindow(target, x, y, width, height, resizable);
        }
        public void SetFollowWindow(bool follow)
        {
            _.SetFollowWindow(follow);
        }

        public void RepairSetting()
        {
            _.RepairSetting();
        }
        public void SaveSetting(string setting)
        {
            _.SaveSetting(setting);
        }
        public void SetVideoExts(string exts)
        {
            _.SetVideoExts(exts);
        }
        public void SetPlayer(string dll, string exe, bool withRun, bool useMove)
        {
            _.SetPlayer(dll, exe, withRun, useMove);
        }

        public void Save(int tab, string text, string path, int type)
        {
            _.Save(tab, text, path, type);
        }
        public void SaveTemp(string text, string path)
        {
            _.SaveTemp(text, path);
        }
        public void OpenFile()
        {
            _.OpenFile();
        }
        public void AfterInit(int limit)
        {
            _.AfterInit(limit);
        }
        public void OpenFileForVideo()
        {
            _.OpenFileForVideo();
        }
        public void SetPath(string smiPath)
        {
            _.SetPath(smiPath);
        }
        public void SetPathAndCheckLoadVideoFile(string smiPath)
        {
            _.SetPath(smiPath);
            _.CheckLoadVideoFile();
        }
        public void LoadVideoFile(string path)
        {
            _.LoadVideoFile(path);
        }
        public void RequestFrames(string path)
        {
            _.RequestFrames(path);
        }
        public void SetThumbnailSize(int width, int height)
        {
            _.SetThumbnailSize(width, height);
        }
        public void RenderThumbnails(string path, string paramsStr)
        {
            _.RenderThumbnails(path, paramsStr);
        }
        public void CancelRenderThumbnails()
        {
            _.CancelRenderThumbnails();
        }
        public void DoExit(bool resetPlayer, bool exitPlayer)
        {
            _.DoExit(resetPlayer, exitPlayer);
        }

        public void Log(string msg)
        {
            // TODO: 로그 파일 저장 따로 만들어야 하나?
            _.Log(msg);
        }
        
        #region 팝업 통신
        // 이거 결국 finder 말곤 안 쓰나?
        public void SendMsg(string target, string msg) {
            _.SendMsg(target, msg);
        }
        // setting.html
        public void GetWindows(string targets) { _.GetWindows(ParseJson<string[]>(targets)); }
        public void SelectPlayerPath() { _.SelectPlayerPath(); }

        // addon 설정용
        public void SetAfterInitAddon(string func) { _.SetAfterInitAddon(func); }
        public void LoadAddonSetting(string path) { _.LoadAddonSetting(path); }
        public void SaveAddonSetting(string path, string text) { _.SaveAddonSetting(path, text); }

        public void GetSubDirs(string dir) { _.GetSubDirs(dir); }
        public void SearchFiles(string dir, string query) { _.SearchFiles(dir, query); }

        // viewer/finder opener 못 쓰게 될 경우 고려
        public void UpdateViewerSetting() { _.UpdateViewerSetting(); }
        public void UpdateViewerLines(string lines) { _.UpdateViewerLines(lines); }

        public void OnloadFinder (string last ) { _.OnloadFinder (last ); }
        public void RunFind      (string param) { _.RunFind      (param); }
        public void RunReplace   (string param) { _.RunReplace   (param); }
        public void RunReplaceAll(string param) { _.RunReplaceAll(param); }
        #endregion

        #region 플레이어
        public void PlayOrPause() { _.player.PlayOrPause(); }
        public void Play() { _.player.Play(); }
        public void Stop() { _.player.Stop(); }
        public void MoveTo(int time) { _.player.MoveTo(time); }
        #endregion

        #region 부가기능
        public void RunColorPicker()
        {
            _.RunColorPicker();
        }
        #endregion
    }
}